persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -348,7 +348,12 @@ protected void loginInternally(io.stargate.db.AuthenticatedUser user) {
 
     private Single<QueryState> newQueryState() {
       if (clientState.getUser() == null) {
-        return Single.just(new QueryState(clientState, UserRolesAndPermissions.ANONYMOUS));
+        // This is here to maintain consistent behavior with C* 3.11 and C* 4.0 since DSE requires
```

Comment:
What if someone from the outside will know this `anonymous` user name?

Reply:
The only time there's no user on the clientState is during internal calls, elsewhere all requests go through an authentication (e.g. cql, rest, and graphql).

Reply:
So there is no way for a requester to log in as `ANONYMOUS_USER`? (does the role have `with login = false` or smth)

Reply:
Well if they somehow logged in with the anonymous user then the clientState would have a user and you wouldn't drop into this block.